wfe2/wfe.go

```diff
@@ -1085,31 +1089,57 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(
 	response.WriteHeader(http.StatusOK)
 }
 
-// Challenge handles POST requests to challenge URLs.
+// ChallengeHandler handles POST requests to challenge URLs of the form /acme/chall-v3/<authorizationID>/<challengeID>.
 // Such requests are clients' responses to the server's challenges.
-func (wfe *WebFrontEndImpl) Challenge(
+func (wfe *WebFrontEndImpl) ChallengeHandler(
 	ctx context.Context,
 	logEvent *web.RequestEvent,
 	response http.ResponseWriter,
 	request *http.Request) {
-	notFound := func() {
+	slug := strings.Split(request.URL.Path, "/")
+	if len(slug) != 2 {
 		wfe.sendError(response, logEvent, probs.NotFound("No such challenge"), nil)
+		return
 	}
+
+	wfe.Challenge(ctx, logEvent, challengePath, response, request, slug[0], slug[1])
+}
+
+// ChallengeHandlerWithAccount handles POST requests to challenge URLs of the form /acme/chall/{regID}/{authzID}/{challID}.
+//
+// [pattern]: https://pkg.go.dev/net/http#hdr-Patterns
```

Comment:
nit: nothing in the comment references this footnote, and we're not actually using the new pattern-handling syntax yet.